src/api/tasks/TaskCollaborators.js

```diff
@@ -149,6 +151,50 @@ class TaskCollaborators extends TasksBase {
         });
     }
 
+    getUrlForNotificationAnalytics(): string {
+        return `${this.getBaseApiUrl()}/v3/notificationAnalyticsEvents`;
+    }
+
+    trackNotificationAnalytics({
+        fileId,
+        userId,
+        taskStatus,
+        taskId,
+        emailTraceId,
+        successCallback,
+        errorCallback
+    }): {
+
+        const url = getUrlForNotificationAnalytics();
+        const requestData = {
+            data: {           
+              notification_type: "workflow-task-" + taskStatus.toLowerCase(),
+              source: 'WEBAPP',
```

Comment:
It seems like this logic should probably live in the webapp, since this component can be used in third-party contexts, as well.